### ISSUE TITLE
[test] test_na_train error: 'MIO_BN_GFX110X' is not defined

### ISSUE THREAD

**Issue by junliume** [ROCmSoftwarePlatform/MIOpen#2472]
@DrizztDoUrden After #2388 the following test starts to fail in CI pipelines.

It is interesting:
- Why it happens now?
- It seems that from the command line `MIO_BN_GFX103X` is defined while `MIO_BN_GFX110X` is not. Maybe we missed it?

```
          Start   39: test_na_train
  38/1533 Test   #39: test_na_train ..................................................................................................................................................................................................................................................................................................................................................................***Failed    1.34 sec
PRNG seed: 12345678
MIOpen(HIP): Info [get_device_name] Raw device name: gfx906:sramecc+:xnack-
MIOpen(HIP): Info [Handle] stream: 0x2456b20, device_id: 3
MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv3, HIP version 5.7.23365, MIOpen version 3.00.0.5798751b2
MIOpen(HIP): Info [Measure] RamDb::Prefetch time: 0.105626 ms
MIOpen(HIP): Info [TryLoad] Find-db regenerating.
MIOpen(HIP): Warning [hip_mem_get_info_wrapper] hipMemGetInfo error, status: 1
MIOpen(HIP): Warning [hip_mem_get_info_wrapper] hipMemGetInfo error, status: 1
MIOpen(HIP): Warning [hip_mem_get_info_wrapper] hipMemGetInfo error, status: 1
MIOpen(HIP): Warning [hip_mem_get_info_wrapper] hipMemGetInfo error, status: 1
MIOpen(HIP): Warning [hip_mem_get_info_wrapper] hipMemGetInfo error, status: 1
MIOpen(HIP): Warning [hip_mem_get_info_wrapper] hipMemGetInfo error, status: 1
MIOpen(HIP): Warning [hip_mem_get_info_wrapper] hipMemGetInfo error, status: 1
MIOpen(HIP): Info [AllocateBuffersAndMakeFusionInvokeParams] in addr: 0x7f873bc06000, size: 802816
MIOpen(HIP): Warning [hip_mem_get_info_wrapper] hipMemGetInfo error, status: 1
MIOpen(HIP): Info [AllocateBuffersAndMakeFusionInvokeParams] out addr: 0x7f873bcca000, size: 802816
MIOpen(HIP): Info [GetFindModeValueImpl] MIOPEN_FIND_MODE = DYNAMIC_HYBRID(5)
MIOpen(HIP): Info [FindSolutionImpl] BnFwdTrgActivationFused (not searchable)
MIOpenBatchNormActivFwdTrainSpatial.cl:36:48: error: 'MIO_BN_GFX110X' is not defined, evaluates to 0 [-Werror,-Wundef]
#if defined(__AMDGCN__) && !(MIO_BN_GFX103X || MIO_BN_GFX110X)
                                               ^
1 error generated.
terminate called after throwing an instance of 'miopen::Exception'
  what():  /var/jenkins/workspace/MLLibs_MIOpen_develop/src/tmp_dir.cpp:45: Can't execute cd /tmp/miopen-MIOpenBatchNormActivFwdTrainSpatial.cl-0d7e-3051-a41d-b6d1; /opt/rocm/llvm/bin/clang -DMIO_BN_N=4 -DMIO_BN_C=64 -DMIO_BN_HW=784 -DMIO_BN_NHW=3136 -DMIO_BN_CHW=50176 -DMIO_BN_NCHW=200704 -DMIO_BN_GRP0=832 -DMIO_BN_GRP1=1 -DMIO_BN_GRP2=1 -DMIO_BN_LDS_SIZE=832 -DMIO_BN_LDSGCN_SIZE=13 -DMIOPEN_READ_UNIT=4 -DMIOPEN_READ_TYPE=_FLOAT4 -DMIO_SAVE_MEAN_VARIANCE=1 -DMIO_RUNNING_RESULT=1 -DMIO_BN_VARIANT=3 -DMIO_BN_GFX103X=0 -DMIOPEN_YES_ACTIV=1 -DMIOPEN_NRN_OP_ID=3 -DMIOPEN_USE_FP16=0 -DMIOPEN_USE_FP32=1 -DSPATIAL_BN -mcpu=gfx906 -Werror -Weverything -Wno-cast-align -Wno-cast-qual -Wno-conversion -Wno-double-promotion -Wno-float-equal -Wno-missing-prototypes -Wno-pass-failed -Wno-pedantic-core-features -Wno-reserved-id-macro -Wno-shorten-64-to-32 -Wno-sign-compare -Wno-sign-conversion -Wno-unused-function -Wno-unused-macros -Wno-declaration-after-statement -mcode-object-version=4 -target amdgcn-amd-amdhsa -x cl -D__AMD__=1  -O3 -cl-kernel-arg-info -cl-denorms-are-zero -cl-std=CL1.2 -mllvm -amdgpu-early-inline-all -mllvm -amdgpu-internalize-symbols  MIOpenBatchNormActivFwdTrainSpatial.cl -o /tmp/miopen-MIOpenBatchNormActivFwdTrainSpatial.cl-0d7e-3051-a41d-b6d1/MIOpenBatchNormActivFwdTrainSpatial.cl.o
CMake Error at test_test_na_train.cmake:7 (message):
  Test failed
```

**Comment by junliume** [comment]
The passing and failing logs are actually more confusing **it should not have passed before #2388**

[pass.log](https://github.com/ROCmSoftwarePlatform/MIOpen/files/13062404/pass.log)
[fail.log](https://github.com/ROCmSoftwarePlatform/MIOpen/files/13062405/fail.log)

Here is the suspicious code snippet:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/5798751b2b4c26d96a867e935d0b768e3d77b70f/src/kernels/MIOpenBatchNormActivBwdSpatial.cl#L34-L40

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/5798751b2b4c26d96a867e935d0b768e3d77b70f/src/kernels/MIOpenBatchNormActivFwdTrainSpatial.cl#L35-L41

The `batchnorm_functions.h` actaully protects if `MIO_BN_GFX110X` is not defined:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/5798751b2b4c26d96a867e935d0b768e3d77b70f/src/kernels/batchnorm_functions.h#L164-L166

However, in the first two files `MIOpenBatchNormActivBwdSpatial.cl` and `MIOpenBatchNormActivFwdTrainSpatial.cl` where `MIO_BN_GFX110X` is used while undefined, the error messages actually make sense:
```
MIOpenBatchNormActivFwdTrainSpatial.cl:36:48: error: 'MIO_BN_GFX110X' is not defined, evaluates to 0 [-Werror,-Wundef]
#if defined(__AMDGCN__) && !(MIO_BN_GFX103X || MIO_BN_GFX110X)
                                               ^
1 error generated.
```
What does not make sense is why the previous `passing.log` shows the test has passed even with the same error in the log.

**Comment by junliume** [comment]
Found some quick fix as mentioned in #2473 by comparing `MIO_BN_GFX103X` and `MIO_BN_GFX110X`.

It seems that #2388 has uncovered a hidden bug in the test procedure (where it should have failed before).

However I do agree with this line:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/5798751b2b4c26d96a867e935d0b768e3d77b70f/src/kernels/batchnorm_functions.h#L132

CC: @JehandadKhan  @atamazov
